config/src/test/java/org/springframework/security/config/http/ClientRegistrationsBeanDefinitionParserTests.java

```diff
@@ -0,0 +1,110 @@
+		assertThat(googleLogin.getClientAuthenticationMethod()).isEqualTo(ClientAuthenticationMethod.BASIC);
+		assertThat(googleLogin.getAuthorizationGrantType()).isEqualTo(AuthorizationGrantType.AUTHORIZATION_CODE);
+		assertThat(googleLogin.getRedirectUriTemplate()).isEqualTo("{baseUrl}/login/oauth2/code/{registrationId}");
+		assertThat(googleLogin.getScopes())
+				.isEqualTo(Sets.newLinkedHashSet(Arrays.asList("openid,profile,email".split(","))));
+		assertThat(googleLogin.getClientName()).isEqualTo("Google");
+
+		ProviderDetails googleProviderDetails = googleLogin.getProviderDetails();
+		assertThat(googleProviderDetails).isNotNull();
+		assertThat(googleProviderDetails.getAuthorizationUri())
+				.isEqualTo("https://accounts.google.com/o/oauth2/v2/auth");
+		assertThat(googleProviderDetails.getTokenUri()).isEqualTo("https://www.googleapis.com/oauth2/v4/token");
+		assertThat(googleProviderDetails.getUserInfoEndpoint().getUri())
+				.isEqualTo("https://www.googleapis.com/oauth2/v3/userinfo");
+		assertThat(googleProviderDetails.getUserInfoEndpoint().getAuthenticationMethod())
+				.isEqualTo(AuthenticationMethod.HEADER);
+		assertThat(googleProviderDetails.getUserInfoEndpoint().getUserNameAttributeName()).isEqualTo("sub");
+		assertThat(googleProviderDetails.getJwkSetUri()).isEqualTo("https://www.googleapis.com/oauth2/v3/certs");
+
+		ClientRegistration githubLogin = clientRegistrationRepository.findByRegistrationId("github-login");
+		assertThat(githubLogin).isNotNull();
+		assertThat(githubLogin.getRegistrationId()).isEqualTo("github-login");
+		assertThat(githubLogin.getClientId()).isEqualTo("github-client-id");
+		assertThat(githubLogin.getClientSecret()).isEqualTo("github-client-secret");
+		assertThat(githubLogin.getClientAuthenticationMethod()).isEqualTo(ClientAuthenticationMethod.BASIC);
+		assertThat(githubLogin.getAuthorizationGrantType()).isEqualTo(AuthorizationGrantType.AUTHORIZATION_CODE);
+		assertThat(githubLogin.getRedirectUriTemplate()).isEqualTo("{baseUrl}/login/oauth2/code/{registrationId}");
+		assertThat(googleLogin.getScopes())
+				.isEqualTo(Sets.newLinkedHashSet(Arrays.asList("openid,profile,email".split(","))));
+		assertThat(githubLogin.getClientName()).isEqualTo("Github");
+
+		ProviderDetails githubProviderDetails = githubLogin.getProviderDetails();
+		assertThat(githubProviderDetails).isNotNull();
+		assertThat(githubProviderDetails.getAuthorizationUri()).isEqualTo("https://github.com/login/oauth/authorize");
+		assertThat(githubProviderDetails.getTokenUri()).isEqualTo("https://github.com/login/oauth/access_token");
+		assertThat(githubProviderDetails.getUserInfoEndpoint().getUri()).isEqualTo("https://api.github.com/user");
+		assertThat(githubProviderDetails.getUserInfoEndpoint().getAuthenticationMethod())
+				.isEqualTo(AuthenticationMethod.HEADER);
+		assertThat(githubProviderDetails.getUserInfoEndpoint().getUserNameAttributeName()).isEqualTo("id");
+	}
+
```

Comment:
Please also add a test for `issuer-uri` to trigger `ClientRegistrations.fromIssuerLocation()` . You could mock the OIDC Discovery Endpoint using `MockWebServer` - see `ClientRegistrationsTest` for an example